Dockerfile

```diff
@@ -1,9 +1,18 @@
-FROM drujensen/crystal:0.23.1
+FROM crystallang/crystal:0.23.1
 
-WORKDIR /app/user
+# Install Dependencies
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -qq && apt-get install -y --no-install-recommends libpq-dev libsqlite3-dev libmysqlclient-dev libreadline-dev git curl vim netcat
 
-ADD . /app/user
+# Install Node
+RUN curl -sL https://deb.nodesource.com/setup_6.x | bash -
+RUN apt-get install -y nodejs
 
-RUN crystal deps
+WORKDIR /opt/amber
+
+# Build Amber
+ENV PATH /opt/amber/bin:$PATH
+ADD . /opt/amber
+RUN shards build amber
```

Comment:
This will add the `amber` cli command to the docker image.  I put it under `/opt/amber/bin` and added it to the PATH.  Crystal is under `/opt/crystal-head/bin` in their base image.

Reply:
I don't see the benefit of putting binaries which are core to the function of the docker container out in `/opt/`. The [filesystem standard](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) says that `/opt` is for optional application software. Given that the point of this docker image is to run an amber server, the amber binary isn't optional. Should it live in `/usr/bin` or even `/bin`? Perhaps with the source in `/usr/src/amber`?

Reply:
@robacarp thanks for reviewing this.  

I think this depends on the purpose of the image.  This image is not meant for a production deployment, it's only focused on development.

I consider the crystal compiler and amber cli as development tools to generate an application.  The crystal image is adding their compiler to `/opt` as well and I think that makes sense, but i'm open to debate here.

Reply:
Either way it’s a semantic issue not standing in the way of the image functionality. No sweat. The only thing I could imagine tripping people up is writing a path variable from scratch, but if the primary purpose of the image is development I don’t think that’s likely.